Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -849,6 +849,148 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Resolve Conflict SG", "[.SyncServe
     }
 }
 
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Auto Purge Enabled - Revoke Access - SGColl", "[.SyncServerCollection]") {
+    const string idPrefix = timePrefix();
+    const string docIDstr = idPrefix + "apera-doc1";
+    const string channelIDa = idPrefix + "a";
+    const string channelIDb = idPrefix + "b";
+
+    _authHeader = SGUserAuthHeader;
+
+    // Create a temporary user for this test
+    HTTPStatus status;
+    C4Error error;
+    sendRemoteRequest("POST", "_user", &status, &error, R"({"name":"purgeRevoke","password":"password"})", true);
+    sendRemoteRequest("PUT", "_user/purgeRevoke", &status, &error,
+                      addChannelToJSON("{}", "admin_channels", { channelIDa, channelIDb }), true);
+    REQUIRE(status == HTTPStatus::OK);
+
+    // Setup pull filter:
+    _pullFilter = [](C4CollectionSpec collectionSpec, C4String docID, C4String revID,
+                     C4RevisionFlags flags, FLDict flbody, void *context) {
+        if ((flags & kRevPurged) == kRevPurged) {
+            ((ReplicatorAPITest*)context)->_counter++;
+            Dict body(flbody);
+            CHECK(body.count() == 0);
+        }
+        return true;
+    };
+
+    // One-shot pull setup
+    constexpr size_t collectionCount = 1;
+    const std::array<C4CollectionSpec, collectionCount> collectionSpecs = {
+            Roses
+    };
+    Encoder enc;
+    enc.beginDict();
+    enc.writeKey(C4STR(kC4ReplicatorOptionChannels));
+    enc.beginArray();
+    enc.writeString(channelIDa);
+    enc.writeString(channelIDb);
+    enc.endArray();
+    enc.writeKey(C4STR(kC4ReplicatorOptionAutoPurge));
+    enc.writeBool(true);
+    enc.endDict();
+    AllocedDict opts { enc.finish() };
+
+    std::array<C4Collection*, collectionCount> collections =
+            collectionPreamble(collectionSpecs, "purgeRevoke", "password");
+    std::array<C4ReplicationCollection, collectionCount> replCollections {
+        {{ // three sets of braces? because Xcode
+            collectionSpecs[0], kC4Disabled, kC4OneShot,
+            opts.data(), nullptr, _pullFilter, this
+        }}
+    };
+
+    C4ParamsSetter paramsSetter = [&replCollections, &opts](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = replCollections.size();
+        c4Params.collections = replCollections.data();
+        for(auto it = Dict(opts).begin(); it != Dict(opts).end(); ++it) {
```

Comment:
I don't think you need this. kC4ReplicatorOptionChannels is an option per-collection. You already added it to the collection.

Reply:
It won't do anything, but will be confusing

Reply:
I see, I think some other error before tripped me up and made me think this was necessary, but I just tested the same point which tripped me up before and it's fine now. Was also confusing to me, so I'm glad you pointed it out.